### PR TITLE
Update to winit 0.27.1 and glutin 0.29.0

### PR DESCRIPTION
**Pull request by budde25**
Bumps winit version

### CODE REVIEW

**Comment by emilk** [comment]
Blocked on https://github.com/gfx-rs/wgpu/pull/2918

**Comment by dbuch** [comment]
This closes #992 btw.

**Comment by emilk** [comment]
A new wgpu release isn’t due until late September, so either we wait until then or we make egui-winit compatible with both old and new winit, with a feature-flag to select which one to use

File: Cargo.lock [hunk 1]
```diff
@@ -1778,12 +1844,6 @@ dependencies = [
...
 
-[[package]]
-name = "glutin_emscripten_sys"
-version = "0.1.1"
-source = "registry+https://github.com/rust-lang/crates.io-index"
-checksum = "80de4146df76e8a6c32b03007bc764ff3249dcaeb4f675d68a06caf1bac363f1"
-
 [[package]]
 name = "glutin_gles2_sys"
 version = "0.1.5"
```
Comment (paulotten) [inline Cargo.lock:1849]: This needs to get bumped to `0.1.6`. See https://github.com/emilk/egui/issues/1896#issuecomment-1209992898 and https://github.com/rust-windowing/glutin/issues/1436

Comment (paulotten) [reply Cargo.lock:1849]: `glutin 0.29.1` fixes this.

Comment (emilk) [reply Cargo.lock:1849]: 0.1.6 hasn't been released yet: https://crates.io/crates/glutin_gles2_sys

Comment (emilk) [reply Cargo.lock:1849]: done!

**Comment by rib** [comment]
I might have a solution to this: https://github.com/rust-windowing/winit/pull/2418

So this way Winit would implement the old and new `raw_window_handle` traits which can hopefully make Winit 0.27 compatible with wgpu 0.13

It would be good to test this egui branch against this

**Comment by emilk** [comment]
> I might have a solution to this: [rust-windowing/winit#2418](https://github.com/rust-windowing/winit/pull/2418)
> 
> So this way Winit would implement the old and new `raw_window_handle` traits which can hopefully make Winit 0.27 compatible with wgpu 0.13
> 
> It would be good to test this egui branch against this

I tested it, and it works! I added this to the root `Cargo.toml`:

```toml
[patch.crates-io]
winit = { git = 'https://github.com/rib/winit/', branch = 'raw_window_handle_04' }
```

and could then run `egui_demo_app` with the `wgpu` feature.

So this PR is now blocked on https://github.com/rust-windowing/winit/pull/2418 plus a winit release.

**Comment by rib** [comment]
ah, cool, thanks for testing/confirming

**Comment by kchibisov** [comment]
Winit 0.27.2 got released.

**Comment by emilk** [comment]
For some reason I can no longer push to this, so I continued the work in https://github.com/emilk/egui/pull/1914

**Pull request closed without merging**